gpii/node_modules/ontologyHandler/src/ontologyHandler.js

```diff
@@ -108,6 +123,156 @@ URLs - this aspect is taken care of in the preferences server component.
+     */
+    gpii.ontologyHandler.floydWarshallAlgorithmInitMatrices = function(data, edges) {
+        // Create |V|x|V| dist and next arrays and initialize to Infinity and null
+        data.dist = fluid.generate(data.numVertices, function() {
+            return fluid.generate(data.numVertices, Infinity, false);
+        }, true);
+        data.next = fluid.generate(data.numVertices, function() {
+            return fluid.generate(data.numVertices, null, false);
+        }, true);
+        fluid.each(edges, function(edge) {
+            var u = data.vertices[edge.source], 
+                v = data.vertices[edge.target];
+            data.dist[u][v] = 1; // All weights currently 1
+            data.next[u][v] = v;
+        });
+    };
+
+    /**
+     * Internal Use only
+     * Function used in gpii.ontologyHandler.floydWarshallAlgorithmWithReconstruction
+     * to perform the core algorithm of Floyd-Warshall.
+     *
+     * @dist (2D array) See algorithm documentation
+     * @next (2D array) See algorithm documentation
+     * @numVertices (Integer) total number of vertices
+     * @returns Does not a return a value, operates directly on the dist and next matrices.
+     */
+    gpii.ontologyHandler.floydWarshallAlgorithmCore = function(dist, next, numVertices) {
+        // Standard Floyd-Warshall implementation
+        for (var k = 0; k < numVertices; k++) {
+            for (var i = 0; i < numVertices; i++) {
+                for (var j = 0; j < numVertices; j++) {
+                    if (dist[i][k] + dist[k][j] < dist[i][j]) {
+                        dist[i][j] = dist[i][k] + dist[k][j];
+                        next[i][j] = next[i][k]
+                    }
+                }
+            }
+        }
+    };
+
+    /*
+     * Uses the data generated in floydWarshallAlgorithmWithReconstruction to
+     * generate the shortest path as an array of vertex names.
+     *
+     * @data (Object) - The data structure returned from 
+     *     floydWarshallAlgorithmWithReconstruction
+     * @v1 (String) - The starting vertice
+     * @v2 (String) - The ending vertice
+     * @returns (Array) - An array of vertices for the shortest path, or an 
+     *     empty array if the path does not exist.
+     */
+    gpii.ontologyHandler.floydWarshallPath = function ( data, v1, v2 ) {
+        if (data.vertices[v1] === undefined || data.vertices[v2] === undefined) {
+            return [];
+        }
+        var path = [],
+            u = data.vertices[v1],
+            v = data.vertices[v2];
+        if (data.next[u][v] === null) {
+            return path;
+        }
+        var getKey = function(val) { 
+            // This hash should have unique values by virtue of it's construction
+            return fluid.find(data.vertices, function(obj, idx) {
+                if (val === obj) {
+                    return idx;
+                }
+            });
+        };
+        path = [getKey(u)];
```

Comment:
We know that getKey(u) must be v1 at this point so just initialise

var path = [v1];  

up front